tests/Mock/Connection/TestBroker.php

```diff
@@ -21,6 +21,8 @@ class TestBroker extends RestClient
 
     protected $apiEndpoint;
 
+    protected $responseData = '{"message": "Some JSON Response Data", "id": "1234"}';
```

Comment:
Hi @TheoKouzelis, IMHO if you're going to override it to test each possible case, then does not make sense (to me) to define it as protected in the parent class. Based on this:
1. A valid option is to remove the parent `protected` property and set it as `private` on the child property.
2. As described above, if we are not going to use it in more than one test (AKA if we don't need to reuse it, or use it more than once) then does not make sense to define it at Class level due it is just used inside one of the functions

Additionally, even when some times `getters` and `setters` are useful, abuse of these methods is not a good idea. You can [read this blog post](http://www.javaworld.com/article/2073723/core-java/why-getter-and-setter-methods-are-evil.html) if you want to know why.